Training/Training/Training.c

```diff
@@ -1,14 +1,194 @@
-// ------------------------------------------------------------------------------------------------
-// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
-#include<stdio.h>
-
-int main () {
-	printf ("Hello World!");
-	return 0;
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// main.c
+// Program to check for palindrome of a string or an integer
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+
+#include<ctype.h>
+#include<errno.h>
+#include <limits.h>
+#include <malloc.h>
+#include<stdio.h>
+#include <stdlib.h>
+#include<string.h>
+
+// Input limit is 200 characters
+#define MAX 202
+#define PALINDROME 1
+#define NOTPALINDROME 0
+#define INVALID -1
+#define ANSI_COLOR_RED     "\x1b[31m"
+#define ANSI_COLOR_GREEN   "\x1b[32m"
+#define ANSI_COLOR_RESET   "\x1b[0m"
+#define ANSI_COLOR_YELLOW  "\x1b[33m"
+
+/// <summary>
+/// Function which checks if a given string is palindrome or not
+/// </summary>

```

Comment:
Summary comments can be in single line.

<summary>Function which checks if a given string is palindrome or not</summary>